include/eve/module/core/function/generic/load.hpp

```diff
@@ -0,0 +1,146 @@
+  }
+
+  template<relative_conditional_expr C, scalar_pointer Ptr>
+  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), C const &cond, Ptr ptr) noexcept
+                  -> as_wide_t<std::remove_cvref_t<decltype(*std::declval<Ptr>())>>
+  {
+    using e_t = std::remove_cvref_t<decltype(*ptr)>;
+    using r_t = as_wide_t< e_t >;
+
+    if constexpr( !std::is_pointer_v<Ptr> )
+    {
+      return eve::load[cond](ptr.get());
+    }
+    else
+    {
+      // If the ignore/keep is complete we can jump over if_else
+      if constexpr( C::is_complete )
+      {
+        if constexpr(C::is_inverted)  { return eve::load(ptr);  }
+        else
+        {
+          if constexpr(C::has_alternative)  return r_t{cond.alternative};
+          else                              return r_t{};
+        }
+      }
+      else
+      {
+        auto offset = cond.offset( as_<r_t>{} );
+
+        if constexpr(C::has_alternative)
+        {
+          [[maybe_unused]] r_t that(cond.alternative);
+          auto* dst   = (e_t*)(&that.storage());
+          std::memcpy( dst + offset, ptr + offset, cond.count( as_<r_t>{} ) );
+          return that;
+        }
+        else
+        {
+          [[maybe_unused]] r_t that;
+          auto* dst   = (e_t*)(&that.storage());
+          std::memcpy( dst + offset, ptr + offset, cond.count( as_<r_t>{} ) );
+          return that;
+        }
+      }
+    }
+  }
+
+  template<relative_conditional_expr C, scalar_pointer Ptr, typename Cardinal>
+  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), C const &cond, Ptr ptr, Cardinal const&) noexcept
+                  -> as_wide_t<std::remove_cvref_t< decltype(*std::declval<Ptr>())>
+                                                  , typename Cardinal::type
+                                                  >
+  {
+    using e_t = std::remove_cvref_t<decltype(*ptr)>;
+    using r_t = as_wide_t< e_t, typename Cardinal::type >;
+
+    if constexpr( !std::is_pointer_v<Ptr> )
+    {
+      return eve::load[cond](ptr.get(), Cardinal{});
+    }
+    else
+    {
+      // If the ignore/keep is complete we can jump over if_else
+      if constexpr( C::is_complete )
+      {
+        if constexpr(C::is_inverted)  { return eve::load(ptr, Cardinal{});  }
+        else
+        {
+          if constexpr(C::has_alternative)  return r_t{cond.alternative};
+          else                              return r_t{};
+        }
+      }
+      else
+      {
+        auto offset = cond.offset( as_<r_t>{} );
+
+        if constexpr(C::has_alternative)
+        {
+          [[maybe_unused]] r_t that(cond.alternative);
+          auto* dst   = (e_t*)(&that.storage());
+          std::memcpy( dst + offset, ptr + offset, cond.count( as_<r_t>{} ) );
+          return that;
+        }
+        else
+        {
+          [[maybe_unused]] r_t that;
+          auto* dst   = (e_t*)(&that.storage());
+          std::memcpy( dst + offset, ptr + offset, cond.count( as_<r_t>{} ) );
+          return that;
+        }
```

Comment:
Aren't your `if/else` branches exactly the same? Why do you need them?

Reply:
the first one fill the return value with the alternative default value befor loading. 
However, this can be factorized into simpler code.

Reply:
Except it doesn't work. I was trying to merge the constexpr if into a local lambda but I am forced to initialize if no alternative are present, thus making a useless pxor.